phylo/src/substitution_models/mod.rs

```diff
@@ -231,8 +315,23 @@ pub struct SubstitutionModelInfo<const N: usize> {
     leaf_sequence_info: Vec<DMatrix<f64>>,
 }
 
-impl<const N: usize> EvolutionaryModelInfo<N> for SubstitutionModelInfo<N> {
-    fn new(info: &PhyloInfo, model: &dyn EvolutionaryModel<N>) -> Result<Self> {
+impl<SubstModel: SubstitutionModel> EvoModelInfo for SubstModelInfo<SubstModel> {
+    type Model = SubstModel;
+
+    fn new(info: &PhyloInfo, model: &SubstModel) -> Result<Self>
+    where
+        Self: Sized,
+    {
+        Self::new(info, model)
+    }
+
+    fn reset(&mut self) {
+        self.reset();
```

Comment:
this feels like a recursion. if not, it needs a comment. otherwise it should be fixed.

Reply:
This wasn't recursing because it was calling an internal implementation, but it was completely unnecessary nonetheless. Removed.